### PR TITLE
Added AttachThreadInput to User32 namespace.

### PR DESCRIPTION
**Pull request by BarryNolte**
Added the DllImport and api signature to the User32 namespace.

### CODE REVIEW

File: src/User32/User32.cs [hunk 1]
```diff
@@ -246,6 +246,19 @@ public static partial class User32
...
+        /// <param name="idAttachTo">The identifier of the thread to which idAttach will be attached.This thread cannot be a system thread. A thread cannot attach to itself. Therefore, idAttachTo cannot equal idAttach.</param>
+        /// <param name="fAttach">If this parameter is TRUE, the two threads are attached.If the parameter is FALSE, the threads are detached.</param>
+        /// <returns>If the function succeeds, the return value is nonzero. If the function fails, the return value is zero.To get extended error information, call GetLastError. Windows Server2003 and WindowsXP:  There is no extended error information; do not call GetLastError.This behavior changed as of WindowsVista.</returns>
+        // [DllImport(nameof(User32), SetLastError = true)]
+        // public static extern bool AttachThreadInput(int idAttach, int idAttachTo, bool fAttach);
+        [DllImport("user32.dll", EntryPoint = "AttachThreadInput")]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static extern bool AttachThreadInput(uint idAttach, uint idAttachTo, [MarshalAs(UnmanagedType.Bool)] bool fAttach);
```
Comment (vbfox) [inline src/User32/User32.cs]: We don't use `uint` in type signatures prefering to always expose the signed type (so `int` here)

File: src/User32/User32.cs [hunk 2]
```diff
@@ -246,6 +246,19 @@ public static partial class User32
...
+        /// </summary>
+        /// <param name="idAttach">The identifier of the thread to be attached to another thread.The thread to be attached cannot be a system thread.</param>
+        /// <param name="idAttachTo">The identifier of the thread to which idAttach will be attached.This thread cannot be a system thread. A thread cannot attach to itself. Therefore, idAttachTo cannot equal idAttach.</param>
+        /// <param name="fAttach">If this parameter is TRUE, the two threads are attached.If the parameter is FALSE, the threads are detached.</param>
+        /// <returns>If the function succeeds, the return value is nonzero. If the function fails, the return value is zero.To get extended error information, call GetLastError. Windows Server2003 and WindowsXP:  There is no extended error information; do not call GetLastError.This behavior changed as of WindowsVista.</returns>
+        // [DllImport(nameof(User32), SetLastError = true)]
+        // public static extern bool AttachThreadInput(int idAttach, int idAttachTo, bool fAttach);
+        [DllImport("user32.dll", EntryPoint = "AttachThreadInput")]
```
Comment (vbfox) [inline src/User32/User32.cs]: `"user32.dll"` should be `nameof(User32)`, no need to specify the `EntryPoint` and the function use `SetLastError` for it's error signaling so it should have `, SetLastError = true`.

**Review by vbfox — CHANGES_REQUESTED** [review]

**Comment by vbfox** [comment]
Thanks for the PR !

Just a few formatting fixed and it's all good for me 😉

File: src/User32/User32.cs [hunk 3]
```diff
@@ -246,6 +246,19 @@ public static partial class User32
...
         public static extern int GetWindowThreadProcessId(IntPtr hWnd, out int lpdwProcessId);
 
+        /// <summary>
+        /// Attaches or detaches the input processing mechanism of one thread to that of another thread.
+        /// </summary>
+        /// <param name="idAttach">The identifier of the thread to be attached to another thread.The thread to be attached cannot be a system thread.</param>
+        /// <param name="idAttachTo">The identifier of the thread to which idAttach will be attached.This thread cannot be a system thread. A thread cannot attach to itself. Therefore, idAttachTo cannot equal idAttach.</param>
+        /// <param name="fAttach">If this parameter is TRUE, the two threads are attached.If the parameter is FALSE, the threads are detached.</param>
+        /// <returns>If the function succeeds, the return value is nonzero. If the function fails, the return value is zero.To get extended error information, call GetLastError. Windows Server2003 and WindowsXP:  There is no extended error information; do not call GetLastError.This behavior changed as of WindowsVista.</returns>
+        // [DllImport(nameof(User32), SetLastError = true)]
```
Comment (vbfox) [inline src/User32/User32.cs]: That declaration and signature actually looks good :)

If only it wasn't commented

**Review by vbfox — CHANGES_REQUESTED** [review]

File: src/User32/User32.cs [hunk 4]
```diff
@@ -246,6 +246,16 @@ public static partial class User32
         [DllImport(nameof(User32), SetLastError = true)]
         public static extern int GetWindowThreadProcessId(IntPtr hWnd, out int lpdwProcessId);
 
+        /// <summary>
+        /// Attaches or detaches the input processing mechanism of one thread to that of another thread.
+        /// </summary>
+        /// <param name="idAttach">The identifier of the thread to be attached to another thread.The thread to be attached cannot be a system thread.</param>
+        /// <param name="idAttachTo">The identifier of the thread to which idAttach will be attached.This thread cannot be a system thread. A thread cannot attach to itself. Therefore, idAttachTo cannot equal idAttach.</param>
+        /// <param name="fAttach">If this parameter is TRUE, the two threads are attached.If the parameter is FALSE, the threads are detached.</param>
+        /// <returns>If the function succeeds, the return value is nonzero. If the function fails, the return value is zero.To get extended error information, call GetLastError. Windows Server2003 and WindowsXP:  There is no extended error information; do not call GetLastError.This behavior changed as of WindowsVista.</returns>
```
Comment (AArnott) [inline src/User32/User32.cs]: You are missing several spaces between words and sentences. Likely this happened when you copied and pasted the MSDN docs and C# "formatted" the text by removing spaces after periods. It's important when you copy and paste that after paste, you immediately hit Undo to revert the formatting changes (but keep the paste).

**Review by vbfox — APPROVED** [review]

**Review by AArnott — APPROVED** [review]
Awesome. Thank you!

**Pull request merged**